xds/internal/balancer/clusterimpl/tests/balancer_test.go

```diff
@@ -170,3 +179,186 @@ func (s) TestConfigUpdateWithSameLoadReportingServerConfig(t *testing.T) {
+	const serviceName = "my-test-xds-service"
+	routeConfigName := "route-" + serviceName
+	clusterName := "cluster-" + serviceName
+	endpointsName := "endpoints-" + serviceName
+	routePolicy := e2e.DefaultRouteConfig(routeConfigName, serviceName, clusterName)
+	// Configure retries as we will shut down server 1 and switch to server 2 later.
+	routePolicy.VirtualHosts[0].RetryPolicy = &v3routepb.RetryPolicy{
+		RetryOn:    "unavailable",
+		NumRetries: &wrapperspb.UInt32Value{Value: 1},
+	}
+	resources := e2e.UpdateOptions{
+		NodeID:    nodeID,
+		Listeners: []*v3listenerpb.Listener{e2e.DefaultClientListener(serviceName, routeConfigName)},
+		Routes:    []*v3routepb.RouteConfiguration{routePolicy},
+		Clusters: []*v3clusterpb.Cluster{
+			{
+				Name:                 clusterName,
+				ClusterDiscoveryType: &v3clusterpb.Cluster_Type{Type: v3clusterpb.Cluster_EDS},
+				EdsClusterConfig: &v3clusterpb.Cluster_EdsClusterConfig{
+					EdsConfig: &v3corepb.ConfigSource{
+						ConfigSourceSpecifier: &v3corepb.ConfigSource_Ads{
+							Ads: &v3corepb.AggregatedConfigSource{},
+						},
+					},
+					ServiceName: endpointsName,
+				},
+				// Specify a custom load balancing policy to use pickfirst.
+				LoadBalancingPolicy: &v3clusterpb.LoadBalancingPolicy{
+					Policies: []*v3clusterpb.LoadBalancingPolicy_Policy{
+						{
+							TypedExtensionConfig: &v3corepb.TypedExtensionConfig{
+								TypedConfig: testutils.MarshalAny(t, &v3pickfirstpb.PickFirst{}),
+							},
+						},
+					},
+				},
+				// Include a fake LRS server config pointing to self.
+				LrsServer: &v3corepb.ConfigSource{
+					ConfigSourceSpecifier: &v3corepb.ConfigSource_Self{
+						Self: &v3corepb.SelfConfigSource{},
+					},
+				},
+			},
+		},
+		Endpoints: []*v3endpointpb.ClusterLoadAssignment{e2e.EndpointResourceWithOptions(e2e.EndpointOptions{
+			ClusterName: endpointsName,
+			Host:        "localhost",
+			Localities: []e2e.LocalityOptions{
+				{
+					Backends: []e2e.BackendOptions{{Port: port1}},
+					Weight:   1,
+				},
+				{
+					Backends: []e2e.BackendOptions{{Port: port2}},
+					Weight:   2,
+				},
+			},
+		})},
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
+	if err := mgmtServer.Update(ctx, resources); err != nil {
+		t.Fatal(err)
+	}
+
+	// Create a ClientConn and make a successful RPC.
+	cc, err := grpc.NewClient(fmt.Sprintf("xds:///%s", serviceName),
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithResolvers(resolverBuilder))
+	if err != nil {
+		t.Fatalf("failed to dial local test server: %v", err)
+	}
+	defer cc.Close()
+
+	client := testgrpc.NewTestServiceClient(cc)
+	var peer peer.Peer
+	if _, err := client.EmptyCall(ctx, &testpb.Empty{}, grpc.Peer(&peer)); err != nil {
+		t.Fatalf("rpc EmptyCall() failed: %v", err)
+	}
+
+	// Verify that the request was sent to server 1.
+	if got, want := testutils.ParsePort(t, peer.Addr.String()), port1; got != want {
```

Comment:
Do we need to parse the port here, can't we just compare `peer.Addr.String()` with `server1.Address`?

Reply:
We don't need to compare ports. Changed to comparing entire addresses.